### PR TITLE
Convert FIRGetOptions to use instance methods (only)

### PR DESCRIPTION
**Pull request by rsgowman**

### CODE REVIEW

File: Firestore/Source/Public/FIRGetOptions.h [hunk 1]
```diff
@@ -65,7 +62,7 @@ typedef NS_ENUM(NSUInteger, FIRGetSource) {
 /**
  * Initializes the get options with the specified source.
  */
-+ (instancetype)optionsWithSource:(FIRGetSource)source NS_SWIFT_NAME(source(_:));
+- (instancetype)optionsWithSource:(FIRGetSource)source NS_SWIFT_NAME(source(_:));
```
Comment (mikelehen) [inline Firestore/Source/Public/FIRGetOptions.h]: `optionsWithSource` suggests immutability.  Perhaps this should be `setSource`?

Comment (morganchen12) [reply Firestore/Source/Public/FIRGetOptions.h]: If it's an initializer, it should be 

```objc
- (instancetype)initWithSource:(FIRGetSource)source
```

The Swift name will automatically be synthesized to `init(source:)`.

**Comment by wilhuff** [comment]
The last commit on here predates our finalizing discussion on this topic. Please update and reassign once it's ready for another pass.

**Comment by rsgowman** [comment]
Further work will be in https://github.com/firebase/firebase-ios-sdk/pull/826. (This PR converts the already-submitted-to-firestore-api-changes-branch-but-not-master work to a style that we've decided we don't want.)

**Pull request closed without merging**